### ISSUE TITLE
[TEAMS] User who isn't in a team can't enter sub-room he's member of (whitepage)

### ISSUE THREAD

**Issue by EnCz** [RocketChat/Rocket.Chat#21573]
### Description:

When a user tries to enter e.g. a private room which is part of a team he's not a member of the application only shows a white page. (Client + Browser)
If someone else writes in that private room the affected user gets a sound effect, so he seems to be a member correctly but it just doesn't get rendered(?).

### Steps to reproduce:

1. Create Team
2. Create private room in team
3. Add second user who isn't member of team to private room
4. Try to enter private room as second user

### Expected behavior:

User sees private room, according to explanation:
Moving a Channel inside a Team means that this Channel will be added in the Team’s context, **however, all Channel’s members, which are not members of the respective Team, will still have access to this Channel**, but will not be added as Team’s members. All Channel’s management will still be made by the owners of this Channel. Team’s members and even Team’s owners, if not a member of this Channel, can not have access to the Channel’s content. Please notice that the Team’s owner will be able remove members from the Channel.

### Actual behavior:

User only sees white page:
![image](https://user-images.githubusercontent.com/7642122/114834354-c2afed80-9dd0-11eb-87c8-869f5ef4c78b.png)

### Server Setup Information:

- Version of Rocket.Chat Server: 3.13.0
- Deployment Method: docker

### Client Setup Information

- Desktop App or Browser Version: Chrome / FireFox
- Operating System: Linux / Windows / Mac

**Comment by Darshilp326** [comment]
Hello! I found out that the private room was not displayed in channels list.

**Comment by yash-rajpal** [comment]
I am not able to replicate this on latest develop branch, everything is working fine for me. 2nd user is able to see and use the private group to which he was added to, after adding this group to team.

**Comment by EnCz** [comment]
@yash-rajpal 
We have this issue on our prod and test instances with different users. (3.13.0)

To verify I pulled up a new server (3.13.0) where I was able to reproduce the problem.

**I then updated the new instance to 3.12.2 and don't have the problem anymore.**
Ticket can be closed, I guess this was fixed in 3.13.1 but not mentioned in the release notes.

Just as an FYI how the problem looked like:
On the left: admin
On the right: user

https://user-images.githubusercontent.com/7642122/114985403-0d923980-9e93-11eb-97f7-9e96119e5640.mp4
